include/RevThread.h

```diff
@@ -0,0 +1,180 @@
+      os << Thread.GetWaitingToJoinTID();
+    } else {
+      os << "N/A";
+    }
+    os << "\n";
+
+    os << '|' << std::string(tableWidth-1, '-') << '|' << '\n';
+    // Table header
+    os << "| " << std::setw(4) << "Reg" << " | " << std::setw(5) << "Alias" << " | " << std::setw(19) << "Value" << " | " << std::setw(21) << "Info" << " |\n";
+    os << "|------|-------|---------------------|-----------------------|\n";
+
+    // Register aliases and descriptions
+    static constexpr const char* aliases[] = {"zero", "ra", "sp", "gp", "tp", "t0", "t1", "t2", "s0", "s1", "a0", "a1", "a2", "a3", "a4", "a5", "a6", "a7", "s2", "s3", "s4", "s5", "s6", "s7", "s8", "s9", "s10", "s11", "t3", "t4", "t5", "t6"};
+    static constexpr const char* info[] = {"Zero Register", "Return Address", "Stack Pointer", "Global Pointer", "Thread Pointer", "Temporary Register", "Temporary Register", "Temporary Register", "Callee Saved Register", "Callee Saved Register", "Arg/Return Register", "Arg/Return Register", "Argument Register", "Argument Register", "Argument Register", "Argument Register", "Argument Register", "Argument Register", "Callee Saved Register", "Callee Saved Register", "Callee Saved Register", "Callee Saved Register", "Callee Saved Register", "Callee Saved Register", "Callee Saved Register", "Callee Saved Register", "Callee Saved Register", "Callee Saved Register", "Temporary Register", "Temporary Register", "Temporary Register", "Temporary Register"};
+
+    // Loop over the registers
+    for (size_t i = 0; i < _REV_NUM_REGS_; ++i) {
+      uint64_t value = RegFile->GetX<uint64_t>(Feature, i);
+      
+      os << "| " << std::setw(4) << ("x" + std::to_string(i));
+      os << " | " << std::setw(5) << aliases[i];
+      os << " | " << std::setw(19) << ("0x" + std::to_string(value));
+      os << " | " << std::setw(21) << info[i] << " |\n";
+    }
+    os << "|" << std::string(tableWidth-1, '-') << "|" << '\n';
+    return os;
+  }
+
+  std::string to_string() const {
+    std::ostringstream oss;
+    oss << *this;      // << operator is overloaded above
+    return oss.str();  // Return the string
+  }
+
+ private:
+  uint32_t ThreadID;                                   // Thread ID
+  uint32_t ParentThreadID;                             // Parent ThreadID
+  uint64_t StackPtr;                                   // Initial stack pointer
+  uint64_t FirstPC;                                    // Initial PC
+  std::shared_ptr<RevMem::MemSegment> ThreadMem;       // TLS and Stack memory
+  RevFeature* Feature;                                 // Feature set for this thread
+  uint64_t ThreadPtr;                                  // Thread pointer
+  ThreadState State = ThreadState::START;              // Thread state (Initializes as )
+  RevRegFile RegFile;                                  // Register file
+  std::unordered_set<uint32_t> ChildrenThreadIDs = {}; // Child thread IDs
+  std::set<int> fildes = {0, 1, 2  };      // Default file descriptors
```

Comment:
`std::unordered_set` would be better than `std::set`  unless you need to traverse the entries in order.

Reply:
Yup Absolutely right!